pele_platform/Frag/libraries.py

```diff
@@ -1,71 +1,119 @@
 
 OUTPUT = "input.conf"
 
-def growing_sites(fragment, user_bond):
+def get_symmetry_groups(mol):
     """
-    Retrieves all possible growing sites (hydrogens) on the fragment. Takes PDB fragment file as input.
-    Output - list of strings represeting sites, e.g. "benzene.pdb C6-H6 C1-H2"
+    Computes the symmetry class for each atom and returns a list with the idx of non-symmetric atoms.
+
+    Parameters
+    ----------
+    mol: Rdkit molecule object.
+
+    Returns
+    -------
+    symmetry_list: List with atom indices.
+    """
+    rank = {} 
+    symmetry_list = []
+    symmetry_rank_list = []
+    counter = 0
+    
+    for counter, atom in enumerate(mol.GetAtoms()):
+        rank[atom.GetIdx()] = list(Chem.CanonicalRankAtoms(mol, breakTies=False))[counter]
+    print(rank)
+    sys.exit() 
+    for idx, symmetry_rank in rank.items():
+        if symmetry_rank not in symmetry_rank_list:
+            symmetry_rank_list.append(symmetry_rank)
+            symmetry_list.append(idx)
+    return symmetry_list
+
+
+def growing_sites(fragment,
+                  user_bond):
     """
-    if hp.is_rdkit():
-        from rdkit import Chem
+    Retrieves all possible growing sites (hydrogens) on the fragment. Takes PDB fragment file as input.
 
+    Parameters
+    ----------
+    fragment: Path to fragment pdb file.
+    user_bond: Connection point from which the user wants to grow the fragments.
+
+    Returns
+    -------
+    bonds: List of strings representing sites, e.g. "benzene.pdb C6-H6 C1-H2"
+    """
     bonds = []
     mol = Chem.MolFromPDBFile(fragment, removeHs=False)
-    
+    symmetry_list = get_symmetry_groups(mol)
     if mol:
         heavy_atoms = [a for a in mol.GetAtoms() if a.GetSymbol() != "H"]
         for a in heavy_atoms:
-            hydrogens = [n for n in a.GetNeighbors() if n.GetSymbol() == "H"]
+            hydrogens = [n for n in a.GetNeighbors() if n.GetSymbol() == "H" and n.GetIdx() in symmetry_list]
             at_name = a.GetMonomerInfo().GetName().strip()
             for h in hydrogens:
                 h_name = h.GetMonomerInfo().GetName().strip()
                 bonds.append("{} {} {}-{}".format(fragment, user_bond, at_name, h_name))
-
     return bonds
 
 
-def sdf_to_pdb(file_list, path, logger):
+def sdf_to_pdb(file_list,
+               logger, tmpdirname):
+    """
+    Converts sdf files to pdb.
 
-    out = []
+    Parameters
+    ----------
+    file_list: List of paths of fragments in sdf format.
+    logger: File with status messages
+    tmpdirname: Path of temporary directory.
 
+    Returns
+    -------
+    out: List of paths with converted sdf files.
+    """
+
+    out = []
     if file_list:
         converted_mae = []
-        output = []
 
         # convert all SDF to MAE
         schrodinger_path = os.path.join(cs.SCHRODINGER, "utilities/structconvert")
         command_mae = "{} -isd {} -omae {}"
         command_pdb = "{} -imae {} -opdb {}"
-
         for f in file_list:
```

Comment:
It would be better to use a variable called `file` instead of `f`. The code below will become more self-explanatory.